src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSectionRootRenderer.cs

```diff
@@ -130,7 +130,19 @@ public override void ViewSafeAreaInsetsDidChange()
 			LayoutHeader();
 		}
 
+		public override void TraitCollectionDidChange(UITraitCollection previousTraitCollection)
+		{
+			base.TraitCollectionDidChange(previousTraitCollection);
+
+			var currentItem = ShellSection.CurrentItem;
 
+			if (!_renderers.TryGetValue(currentItem, out var currentItemHandler))
+				return;
+
+			var application = currentItemHandler.GetRequiredService<IApplication>();
```

Comment:
Can you just get it from `_shellContext` ? 

`_shellContext.Shell.FindMauiContext().Services`